src/libs/actions/User.js

```diff
@@ -558,15 +566,29 @@ function subscribeToUserEvents() {
             // until we finish the migration to reliable updates. So let's check it before actually updating
             // the properties in Onyx
             if (pushJSON.lastUpdateID && pushJSON.previousUpdateID) {
-                console.debug('[OnyxUpdates] Received lastUpdateID from pusher', pushJSON.lastUpdateID);
-                console.debug('[OnyxUpdates] Received previousUpdateID from pusher', pushJSON.previousUpdateID);
-                // Store these values in Onyx to allow App.reconnectApp() to fetch incremental updates from the server when a previous session is being reconnected to.
-                Onyx.multiSet({
-                    [ONYXKEYS.ONYX_UPDATES.LAST_UPDATE_ID]: Number(pushJSON.lastUpdateID || 0),
-                    [ONYXKEYS.ONYX_UPDATES.PREVIOUS_UPDATE_ID]: Number(pushJSON.previousUpdateID || 0),
-                });
+                const pushJSONLastUpdateID = Number(pushJSON.lastUpdateID || 0);
+                const pushJSONPreviousUpdateID = Number(pushJSON.previousUpdateID || 0);
+
+                console.debug('[OnyxUpdates] Received lastUpdateID from pusher', pushJSONLastUpdateID);
+                console.debug('[OnyxUpdates] Received previousUpdateID from pusher', pushJSONPreviousUpdateID);
+                console.debug('[OnyxUpdates] The lastUpdateID the client received was', onyxUpdatesLastUpdateID);
+
+                // Store this value in Onyx to allow AuthScreens to fetch incremental updates from the server when a previous session is being reconnected to.
+                Onyx.set(ONYXKEYS.ONYX_UPDATES_LAST_UPDATE_ID, pushJSONLastUpdateID);
+
+                // The previous update from the server does not match the last update the client got which means the client is missing some updates.
+                // ReconnectApp will fetch updates starting from the last update this client got and going to the last update the server sent.
+                if (onyxUpdatesLastUpdateID < pushJSONPreviousUpdateID) {
+                    console.debug('[OnyxUpdates] Gap detected in update IDs so fetching incremental updates');
+                    Log.info('Gap detected in update IDs from Pusher so fetching incremental updates', true, {
+                        lastUpdateIDFromPusher: pushJSONLastUpdateID,
+                        previousUpdateIDFromPusher: pushJSONPreviousUpdateID,
+                        lastUpdateIDOnClient: onyxUpdatesLastUpdateID,
+                    });
+                    App.reconnectApp(onyxUpdatesLastUpdateID, pushJSONLastUpdateID, true);
+                }
```

Comment:
@tgolen I think you missed this, but you created a [GetMissingOnyxMessages](https://github.com/Expensify/Web-Expensify/blob/503f3fa79bd2862b94ad09f912df0e1e38b2e97b/api.php#L3001-L3003) method in the API to use exactly on this situation instead of using ReconnectApp. I think we should use it here for clarity.

Reply:
OK, I will switch to using this.